### ISSUE TITLE
Support for Hammerwatch 1.4?

### ISSUE THREAD

**Issue by MedicInMirrorshades** [joedf/HWSEdit#5]
Wondering if you'll add support for Hammerwatch 1.4 including the Sorcerer class... doable?

**Comment by joedf** [comment]
Doable. I’m on vacation right now. You can take a jab at it and make pull request in the meantime. You can try out the XML tool for a look at the data structure.

**Comment by darkslayer322** [comment]
Hey, I would also request this.

I have very limited coding experience myself, but i am trying to look at your code and see if i can make it recognize saves again.

**Comment by joedf** [comment]
Nice, I actually forgot about this. Haven't played Hammerwatch in a long time.
You'll want to look at MainForm.cs' Open():
https://github.com/joedf/HWSEdit/blob/44b74e5395fd2e97f280bfc381cf3f97d8496ca6/HWSEdit/MainForm.cs#L459

**Comment by darkslayer322** [comment]
Sadly this was beyond my current knowledge. 

https://github.com/joedf/HWSEdit/blob/44b74e5395fd2e97f280bfc381cf3f97d8496ca6/HWSEdit/hwsFormat.cs#L385-L388

Here it returns 72 no matter which .hws file i choose, which causes it to return null, leaving MAINBUFFER as null 🤷 

Im guessing the load/save code has changed since you got it from the devs

**Comment by joedf** [comment]
Yes, that probably why it's broken... But you ça still use the AHK version, I think it's should still unpack the files.

**Comment by joedf** [comment]
It's been reported to be working with v1.41 ?
https://github.com/joedf/HWSEdit/pull/7#issuecomment-1008879374

**Comment by Mirppc** [comment]
I can confirm it does not like the current version of Heros of Hammerwatch.  Same error about invalid save.

**Comment by joedf** [comment]
Thanks for the info, that's good to know! However, I have not worked on or updated this app for it work with the newer versions of Hammerwatch. I don't own a copy of "Heros of Hammerwatch". Providing a save file might help, and that is if have the time or motivation 😛 

I am open to PRs or forks! :)
